services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java

```diff
@@ -154,7 +153,18 @@
     // used solely to determine is user is logged in
     public static final String KEYCLOAK_SESSION_COOKIE = "KEYCLOAK_SESSION";
     public static final String KEYCLOAK_REMEMBER_ME = "KEYCLOAK_REMEMBER_ME";
+
+    // ** Logout related notes **/
+    // Flag in the logout session to specify if we use "system" client or real client
+    public static final String LOGOUT_WITH_SYSTEM_CLIENT = "LOGOUT_WITH_SYSTEM_CLIENT";
```

Comment:
Not quite following the "system" client scenario. Could you elaborate on this one?

Reply:
I was thinking about removing this constant during the PR cleanup, will probably do that.

The point of the "system" client is, that AuthenticationSessionModel always need to have client, which it is linked too. In normal OIDC/SAML login scenarios, authentication session always has client. But in some rare cases (ActionTokens, or now Logout) there is not dedicated client of the flow and hence the "system" client is used as a placeholder. The system client is currently "account" client.

From the long term perspective, it may be better to remove the need of "system client" or use dedicated client just for that instead of the "account" client.

In the case of the system client, the link "Back to application" should not be shown on the pages as it is not correct to show "Back to application" to point to account management in case that "account" was used only as a placeholder system client. Hence I've added the session note to differentiate between the cases when "account" was used as a "system" client or as real client during logout from account console. However session note is not needed as logout from account console is not a problem because:

- New account console uses "account-console" client
- Old account console uses "account" client, but user is immediately logged out when click to "Logout"

So I can remove the note and instead directly check if the client is the system client (account) to decide if the link "Back to application" should be shown.

Reply:
Updated to remove this session note and instead directly check if client is "system client" to doublecheck if the link "Back to application" should be shown or not.